From eeaefe3656e67ce574f3fe7cf9d5d4ab02626f71 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Tue, 7 May 2019 17:20:06 +0200
Subject: [PATCH 218/345] watchdog: octeon-wdt: Fix timer rate for all OCTEON
 III parts.

All CIU3 based parts have a count specified in multiples of 1024.

Signed-off-by: David Daney <david.daney@cavium.com>
---
 drivers/watchdog/octeon-wdt-main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/watchdog/octeon-wdt-main.c b/drivers/watchdog/octeon-wdt-main.c
index 0ec419a3f7ed..eb495870f2d6 100644
--- a/drivers/watchdog/octeon-wdt-main.c
+++ b/drivers/watchdog/octeon-wdt-main.c
@@ -538,7 +538,7 @@ static int __init octeon_wdt_init(void)
 
 	if (OCTEON_IS_MODEL(OCTEON_CN68XX))
 		divisor = 0x200;
-	else if (OCTEON_IS_MODEL(OCTEON_CN78XX))
+	else if (octeon_has_feature(OCTEON_FEATURE_CIU3))
 		divisor = 0x400;
 	else
 		divisor = 0x100;
-- 
2.25.1

